fourth-challenge/tests/Feature/Cities/CityTest.php

```diff
@@ -0,0 +1,148 @@
+            'name' => 'Miami',
+            'flights_arriving' => 100,
+            'flights_departing' => 100,
+        ]);
+    }
+
+    /** @test */
+    public function index_all_cities()
+    {
+        $this->withoutExceptionHandling();
+
+        City::factory()->count(2)->sequence(
+            ['name' => 'Miami'],
+            ['name' => 'London'],
+        )
+        ->create();
+
+        $this->get('/cities')
+            ->assertSuccessful()
+            ->assertSee('Miami')
+            ->assertSeeInOrder([
+                'Miami',
+                'London',
+            ]);
+    }
+
+/** @test */
+    public function destroy_a_city()
+    {
+        $this->withoutExceptionHandling();
+
+        $city = City::factory()->create();
+
+        $this->delete(action([CityController::class, 'destroy'], $city))
+            ->assertSuccessful();
+
+        $this->assertDatabaseMissing(City::class, [
+            'id' => $city->id,
+        ]);
+    }
+
+/** @test */
+    public function update_a_city()
+    {
+        $this->withoutExceptionHandling();
+
+        $city = City::factory()->create([
+            'name' => 'New York City',
+            'flights_arriving' => 200,
+            'flights_departing' => 200,
+        ]);
+
+        $this->patch(action([CityController::class, 'update'], $city), [
+            'name' => 'Miami',
+            'flights_arriving' => 100,
+            'flights_departing' => 100,
+        ])
+            ->assertSuccessful();
+
+        $this->assertDatabaseHas(City::class, [
+            'id' => $city->id,
+            'name' => 'Miami',
+            'flights_arriving' => 100,
+            'flights_departing' => 100,
+        ])
+            ->assertDatabaseMissing(City::class, [
+                'id' => $city->id,
+                'name' => 'New York City',
+                'flights_arriving' => 200,
+                'flights_departing' => 200,
+            ]);
+    }
+
+    /** @test */
+    public function validate_a_city_fields(){
+
+        $city = City::factory()->create([
+            'name' => 'New York City',
+            'flights_arriving' => 200,
+            'flights_departing' => 200,
+        ]);
+
+        $this->patch(action([CityController::class, 'update'], $city), [
+            'name' => '',
+            'flights_arriving' => 100,
+            'flights_departing' => 100,
+        ])
+            ->assertSessionHasErrors('name');
+
+        $this->patch(action([CityController::class, 'update'], $city), [
+            'name' => 'Miami',
+            'flights_arriving' => '',
+            'flights_departing' => 100,
+        ])
+            ->assertSessionHasErrors('flights_arriving');
+
+        $this->patch(action([CityController::class, 'update'], $city), [
+            'name' => 'Miami',
+            'flights_arriving' => 100,
+            'flights_departing' => '',
+        ])
+            ->assertSessionHasErrors('flights_departing');
```

Comment:
Para este tipo de tests que lo que haces en fin es repetir la misma assertion con pequeños cambios en los body de los requests, phpunit tiene lo que son dataProviders (https://phpunit.de/manual/3.7/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers). Esta funcionalidad te permite escribir un solo test y despues especificar una lista de distintos escenarios con los que queres que se corra, te sirve para evitar duplicar codigo, te queda mas prolijo y en fin dejas solo una assertion por test que es mejor practica